src/Model/Repository/BaseRepository.php

```diff
@@ -0,0 +1,116 @@
+<?php
+
+
+namespace Friendica\Model\Repository;
+
+use Friendica\Database\Database;
+use Friendica\Model\Entity\BaseEntity;
+use Friendica\Model\Entity\IStorable;
+
+abstract class BaseRepository
+{
+	/** @var string */
+	protected static $table;
+	/** @var BaseEntity */
+	protected static $entity;
+
+	/** @var Database */
+	protected $dba;
+
+	/**
+	 * BaseRepository constructor.
+	 *
+	 * @param Database $dba
+	 */
+	public function __construct(Database $dba)
+	{
+		$this->dba = $dba;
+	}
+
+	/**
+	 * @param array $data
+	 *
+	 * @return mixed
+	 */
+	abstract protected function createEntity(array $data);
+
+	/**
+	 * @param array $conditions
+	 *
+	 * @return mixed
+	 *
+	 * @throws \Exception
+	 */
+	public function select(array $conditions)
+	{
+		$prototype = null;
+
+		$selected = $this->dba->delete(static::$table, [], $conditions);
+
+		$return = [];
+
+		while ($data = $this->dba->fetch($selected)) {
+			if ($prototype === null) {
+				$prototype = new static::$entity($data);
+				$return[] = $prototype;
+			} else {
+				$return[] = static::$entity::createFromPrototype($prototype, $data);
+			}
+		}
+
+		return $data;
+	}
+
+	/**
+	 * @param IStorable[] $entities
+	 *
+	 * @return boolean True, if deleted
+	 */
+	public function delete(IStorable ...$entities)
```

Comment:
This one is a generic way to delete 1 ... n entities. They just need to implement the `IStorable` interface

Reply:
Why not take an array of entities, or even better, a collection!

Reply:
I thought so we can pass one entity as well as a list of it. I hadn't tried it, but wouldn't `DI::repo()->delete($entitiyArray);` work as well?